assignments/array-methods.js

```diff
@@ -56,28 +56,63 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(runner){
+    fullName.push(`${runner.first_name}, ${runner.last_name}`);//IF THIS IS CORRECT I CANT BELIEVE I FIGURED IT OUT ON THE FIRST TRY!!!!!
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+const makeThemUpper = runners.map(function(runner){
+    allCaps.push(runner.first_name.toUpperCase())
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+const shirtSizeFilter = runners.filter(function(runner){
+    if(runner.shirt_size === "L"){
+        largeShirts.push(runner)
+    }
```

Comment:
This works similarly to map. The filter array method:

* Creates a new array
* Calls the callback function on each element on the old array
  * It expects a true/false from the callback function
    * If true, it pushes the element into the new array
    * If false, it leaves it alone
* Returns the new array

Therefore, you can just simply write it like this:

```js
const largeShirts = runners.filter(function(runner) {
    if (runner.shirt_size === "L") {
        return true;
    } else {
        return false;
    }
    /* Alternatively, simply:
     * return runner.shirt_size === "L";
     */
});
```

Do `console.log(largeShirts);`. You'll find the result to be the same.